pkg/postgres/query/compose.go

```diff
@@ -37,6 +37,40 @@ func composeQuery(
 		conditionTemplate, err = simpleOperatorCondition(
 			field, valueIsList, " %s != ?", " %s NOT IN (%s)",
 		)
+	case filter.CompareOperatorIsLessThan:
+		conditionTemplate, err = simpleOperatorCondition(
+			field, valueIsList, " %s < ?", " %s < LEAST(%s)",
+		)
+	case filter.CompareOperatorIsLessThanOrEqualTo:
+		conditionTemplate, err = simpleOperatorCondition(
+			field, valueIsList, " %s <= ?", " %s <= LEAST(%s)",
+		)
+	case filter.CompareOperatorIsGreaterThan:
+		conditionTemplate, err = simpleOperatorCondition(
+			field, valueIsList, " %s > ?", " %s > GREATEST(%s)",
+		)
+	case filter.CompareOperatorIsGreaterThanOrEqualTo:
+		conditionTemplate, err = simpleOperatorCondition(
+			field, valueIsList, " %s >= ?", " %s >= GREATEST(%s)",
+		)
+	case filter.CompareOperatorContains:
+		conditionTemplate, err = likeOperatorCondition(
+			field, valueIsList, false, false,
```

Comment:
Asset escapes special symbols (see `likeReplacer` in the code there ) for the arguments when using an `ILIKE` operator. I assume we need to do that as well, or does sqlx handle the escaping ?

E.g. a `BeginsWIth:B%SI` in the resultSelector should match for all strings starting with `B%SI` (cas insensitive) but not for e.g. `Buttersilo` (unescaped the `%` in the passed argument would match `utter` here)

Reply:
Not sure if `sqlx` handles escaping, but the [documentation](https://jmoiron.github.io/sqlx/#:~:text=the%20RETURNING%20clause.-,bindvars,-%C2%B6) says that `bindvars` prevents SQL Injection _(which could possible mean handling escaping special symbols)_

However, I adjust the `extractFieldValues` to handle escaping special symbols on specific cases before appending to the args list to be used by `sqlx`
Checkout this new implementation extractFieldValues